### PR TITLE
Move FOCS file reading to Directory code

### PR DESCRIPTION
**Pull request by o01eg**
Should simplify access to data unavailable via file operations like Android assets #3202 or use it indirection to implement modding support #3204.

### CODE REVIEW

File: util/StringTable.cpp [hunk 1]
```diff
@@ -57,7 +57,7 @@ void StringTable::Load(std::shared_ptr<const StringTable> fallback) {
     auto path = FilenameToPath(m_filename);
     std::string file_contents;
 
-    bool read_success = parse::read_file(path, file_contents);
+    bool read_success = ReadFOCScript(path, file_contents);
```
Comment (geoffthemedio) [inline util/StringTable.cpp]: A stringtable isn't FOCS

File: util/Directories.cpp [hunk 2]
```diff
@@ -734,3 +734,25 @@ auto IsExistingFile(const fs::path& path) -> bool
 
     return false;
 }
+
+auto ReadFOCScript(boost::filesystem::path const& path, std::string& file_contents) -> bool
```
Comment (geoffthemedio) [inline util/Directories.cpp]: this isn't FOCS specific... the old name was more accurate

File: util/StringTable.cpp (hunk 1 above)
Comment (o01eg) [reply util/StringTable.cpp]: Aren't it parsed for FOCS substitutions?

File: util/Directories.cpp (hunk 2 above)
Comment (o01eg) [reply util/Directories.cpp]: OK, changed to `ReadFile`.

**Comment by o01eg** [comment]
I'll remove `ReadFile` from `ShaderProgram`.

File: util/StringTable.cpp (hunk 1 above)
Comment (geoffthemedio) [reply util/StringTable.cpp]: Macro substitutions aren't [FO] content scripts. Even if they were, this function doesn't do them.

Comment (o01eg) [reply util/StringTable.cpp]: OK, I removed FOCS mention here. Moreover I suddenly fixed code duplication in `ShaderProgram`.

**Pull request merged**